### PR TITLE
 DB-8256 allow nestedloop join condition to be pushed down to derived table

### PR DESCRIPTION
**Pull request by yxia92**

### CODE REVIEW

**Comment by yxia92** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: splice_machine/src/test/java/com/splicemachine/db/impl/sql/compile/NLJPredicatePushedToDT.java [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+
+import static com.splicemachine.test_tools.Rows.row;
+import static com.splicemachine.test_tools.Rows.rows;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Created by yxia on 8/19/20.
+ */
+@RunWith(Parameterized.class)
+public class NLJPredicatePushedToDT extends SpliceUnitTest {
```
Comment (dgomezferro) [inline splice_machine/src/test/java/com/splicemachine/db/impl/sql/compile/NLJPredicatePushedToDT.java:45]: Nitpick: can you end the name in *IT to follow the convention?

**Review by dgomezferro — APPROVED** [review]

**Comment by yxia92** [comment]
@dgomezferro I pushed a new commit to prevent nestedloop join from pushing down for window function as it is not semantically equivalent and could lead to wrong result. Please take a look at the last commit. More detail is in the JIRA. Thanks!

**Pull request merged**